Security (#7)
tjav/upskillingquests: Security/Basics/security-basics-1.md

# Security-Basics-1 Role Based Access

## Description

> Constoso wants to ensure that the 'least priviledge access' concept is applied to the Azure environment as well. Contoso want to investigate and learn more abour Role and Access management in Azure. Contoso is considering restricting access to the environment and ensuring that only service principals have write access to the environment, and users only can read access. 

Reviewer: ```suggestion
Contoso wants to ensure that the 'least priviledge access' concept is applied to the Azure environment as well. Contoso want to investigate and learn more about Role and Access management in Azure. Contoso is considering restricting access to the environment and ensuring that only service principals have write access to the environment, and users only can read access. 
```

Reviewer: And "Abou**r**"

> There are specific departmental needs who need to audit or review custom aspects of the cloud.

Reviewer: ```suggestion
There are specific department needs who need to audit or review custom aspects of the cloud.
```

## Outcome

Resource access: Create a Resource Group and a Service Prinipal (app registration) called `App1`. Create a custom role that allows the creation of VMs and VNETs in the `App1` RG for the `App1` service principal. 

> DevOps tema access: Create a group in Azure AD called `App1 DevOps Team`. Grant Reader role for the group on the `App1` Resource Group.

Reviewer: ```suggestion
DevOps team access: Create a group in Azure AD called `App1 DevOps Team`. Grant Reader role for the group on the `App1` Resource Group.
```

Access to the security auditors: Create a group in Azure AD called `Security Auditors`. Grant Security Reader rights for this group in the Azure environment

> Try to leverage scripts/templates/code for implmenting these

Reviewer: ```suggestion
Try to leverage scripts/templates/code for implementing these
```

## Review and discussion
What are the pros and cons for using Custom roles?

> When would yo recommend using custom roles? When would you recommend using built-in roles?

Reviewer: ```suggestion
When would you recommend using custom roles? When would you recommend using built-in roles?
```

What is the difference between Owner and Contributor built in roles? Why is it relevant?
In the scenario: What do you think of granting only Read access for the DevOps team members? Why should or should not the DevOps team have Contributor or Owner rights on their own app?

If you want to talk through or validate your quest after you completed, contact [Matyas Safranka](mailto:matyas@microsoft.com) via email or Teams.

## References

> It is not mandantory to use these references.

Reviewer: ```suggestion
It is not mandatory to use these references.
```
